### PR TITLE
AB#9725 - Removed IUmbracoApplicationLifetimeManager

### PR DESCRIPTION
**Pull request by bergmania**
Fixes [AB#9725](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/9725)

### Details
- Removed `IUmbracoApplicationLifetimeManager`
- Removed `IUmbracoApplicationLifetime.ApplicationInit`
- Removed `AspNetCoreComposer` + `AspNetCoreComponent`
- Migrated `WebProfilerComposer` + `WebProfilerComponent`
  - Added to UmbracoBuilder + Created new `InitializeWebProfiling` notification hander executed on `UmbracoApplicationStarting`
- Fixed issues with web profiler rendering on the frontend (This can't be handled in the same way anymore)

### Test
- Run in debug mode
- Visit https://localhost:44331/?umbdebug=true
  - First request should include the boot profiling
  - Subsequent request should only include the request profiling

### CODE REVIEW

File: src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        private readonly bool _profile;
+        private readonly WebProfiler _profiler;
+        private readonly IUmbracoRequestLifetime _umbracoRequestLifetime;
+        private readonly List<Action> _terminate = new List<Action>();
+        public InitializeWebProfiling(IProfiler profiler, IUmbracoRequestLifetime umbracoRequestLifetime, ILogger<InitializeWebProfiling> logger)
+        {
+            _umbracoRequestLifetime = umbracoRequestLifetime;
+            _profile = true;
+
+            // although registered in WebRuntime.Compose, ensure that we have not
```
Comment (Shazwazza) [inline src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs:24]: Is this still relevant?

File: src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+using Umbraco.Web.Common.Lifetime;
+
+namespace Umbraco.Web.Common.Profiler
+{
+    public class InitializeWebProfiling : INotificationHandler<UmbracoApplicationStarting>
+    {
+        private readonly bool _profile;
+        private readonly WebProfiler _profiler;
+        private readonly IUmbracoRequestLifetime _umbracoRequestLifetime;
+        private readonly List<Action> _terminate = new List<Action>();
```
Comment (Shazwazza) [inline src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs:18]: Adding stuff to _terminate doesn't seem to do anything now

File: src/Umbraco.Tests/Scoping/ScopedNuCacheTests.cs [hunk 3]
```diff
@@ -106,7 +105,7 @@ protected override IPublishedSnapshotService CreatePublishedSnapshotService(Glob
                 hostingEnvironment,
                 Microsoft.Extensions.Options.Options.Create(nuCacheSettings));
 
-            lifetime.Raise(e => e.ApplicationInit += null, EventArgs.Empty);
+            //lifetime.Raise(e => e.ApplicationInit += null, EventArgs.Empty);
```
Comment (Shazwazza) [inline src/Umbraco.Tests/Scoping/ScopedNuCacheTests.cs:108]: can remove comment

File: src/Umbraco.Core/DependencyInjection/UmbracoBuilder.cs [hunk 4]
```diff
@@ -114,9 +114,8 @@ private void AddCoreServices()
             // Adds no-op registrations as many core services require these dependencies but these
             // dependencies cannot be fulfilled in the Core project
             Services.AddUnique<IMarchal, NoopMarchal>();
-            Services.AddUnique<IProfiler, NoopProfiler>();
+            //Services.AddUnique<IProfiler, NoopProfiler>();
```
Comment (Shazwazza) [inline src/Umbraco.Core/DependencyInjection/UmbracoBuilder.cs:117]: can remove commented out code. do we still need NoopProfiler? it's being checked for but doesn't seem like we use/need it?

**Review by Shazwazza — CHANGES_REQUESTED** [review]
Left a few comments. Tests and profiler tests are passing though

File: src/Umbraco.Core/DependencyInjection/UmbracoBuilder.cs (hunk 4 above)
Comment (bergmania) [reply src/Umbraco.Core/DependencyInjection/UmbracoBuilder.cs:117]: We still new up `NoopProfiler` in `Umbraco.Web.Common.DependencyInjection.UmbracoBuilderExtensions.GetWebProfiler`. So it is still needed

File: src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs (hunk 1 above)
Comment (bergmania) [reply src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs:24]: I'll update the comment, but in theory, the profiler can still be replaced by another type before the `UmbracoApplicationStarting` notification is handled

File: src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs (hunk 2 above)
Comment (bergmania) [reply src/Umbraco.Web.Common/Profiler/InitializeWebProfiling.cs:18]: Damn, of course not..  As har as I know, we don't really have a good way to handle this? 
1. We can use the `UmbracoApplicationStopping` event, but then we don't really has the connection between the two notification handlers?
2. We could change the IUmbracoRequestLifetime.RequestStart event to be handled by the EventAggregator.

**Pull request merged**